stat/distuv/statdist.go

```diff
@@ -10,6 +10,69 @@ import (
 	"gonum.org/v1/gonum/mathext"
 )
 
+// Bhattacharyya is a type for computing the Bhattacharyya distance between
+// probability distributions.
+//
+// The Bhattacharyya distance is defined as
+//  D_B = -ln(BC(l,r))
+//  BC = \int_-∞^∞ (p(x)q(x))^(1/2) dx
+// Where BC is known as the Bhattacharyya coefficient.
+// The Bhattacharyya distance is related to the Hellinger distance by
+//  H = sqrt(1-BC)
+// For more information, see
+//  https://en.wikipedia.org/wiki/Bhattacharyya_distance
+type Bhattacharyya struct{}
+
+// DistBeta returns the Bhattacharyya distance between Beta distributions l and r.
+// For Beta distributions, the Bhattacharyya distance is given by
+//  -ln(B((α_l + α_r)/2, (β_l + β_r)/2) / (B(α_l,β_l), B(α_r,β_r)))
+// Where B is the Beta function.
+func (Bhattacharyya) DistBeta(l, r Beta) float64 {
+	// Reference: https://en.wikipedia.org/wiki/Hellinger_distance#Examples
+	return -mathext.Lbeta((l.Alpha+r.Alpha)/2, (l.Beta+r.Beta)/2) +
+		0.5*mathext.Lbeta(l.Alpha, l.Beta) + 0.5*mathext.Lbeta(r.Alpha, r.Beta)
+}
+
+// DistNormal returns the Bhattacharyya distance Normal distributions l and r.
+// For Normal distributions, the Bhattacharyya distance is given by
+//  s = (σ_l^2 + σ_r^2)/2
+//  BC = 1/8 (μ_l-μ_r)^2/s + 1/2 ln(s/(σ_l*σ_r))
+func (Bhattacharyya) DistNormal(l, r Normal) float64 {
+	// Reference: https://en.wikipedia.org/wiki/Bhattacharyya_distance
+	m := l.Mu - r.Mu
+	s := (l.Sigma*l.Sigma + r.Sigma*r.Sigma) / 2
```

Comment:
I think this would be more clear (given the ref to the example at WP) if the division here were removed and done in the expression below, `0.25*m*m/s + 0.5*math.Log(s/2) - 0.5*math.Log(l.Sigma) - 0.5*math.Log(r.Sigma)`. No?

Reply:
I see. I was simplifying the multivariate normal formula (and missed the equation for the single). I personally think the multivariate formula is much clearer what is going on (in that the terms are clearer), but I'm happy to do it either way

Reply:
This is fine.